docs/adrs/0076-partial-update-patches.md

```diff
@@ -0,0 +1,149 @@
+The MTOShipmentUpdate struct would represent an update to an MTOShipment. It would resemble the MTOShipment struct, but its fields could be Nullable, which means that they would contain information about whether a field should be updated by the update service.
+
+```
+type Nullable[T any] struct {
+	Present bool
+	Value   T
+}
+```
+
+```
+type MTOShipmentUpdate struct {
+	ID                               uuid.UUID
+	MoveTaskOrder                    Move
+	MoveTaskOrderID                  uuid.UUID
+	ScheduledPickupDate              *time.Time
+	RequestedPickupDate              *time.Time
+	RequestedDeliveryDate            *time.Time
+	ApprovedDate                     *time.Time
+	FirstAvailableDeliveryDate       *time.Time
+	ActualPickupDate                 *time.Time
+	RequiredDeliveryDate             *time.Time
+	ScheduledDeliveryDate            *time.Time
+	ActualDeliveryDate               *time.Time
+	CustomerRemarks                  *string
+	CounselorRemarks                 Nullable[*string]
+	PickupAddress                    *Address
+	PickupAddressID                  *uuid.UUID
+	DestinationAddress               *Address
+	DestinationAddressID             *uuid.UUID
+	DestinationType                  *DestinationType
+	MTOAgents                        MTOAgents
+	MTOServiceItems                  MTOServiceItems
+	SecondaryPickupAddress           *Address
+	SecondaryPickupAddressID         *uuid.UUID
+	SecondaryDeliveryAddress         *Address
+	SecondaryDeliveryAddressID       *uuid.UUID
+	SITDaysAllowance                 *int
+	SITExtensions                    SITExtensions
+	PrimeEstimatedWeight             *unit.Pound
+	PrimeEstimatedWeightRecordedDate *time.Time
+	PrimeActualWeight                *unit.Pound
+	BillableWeightCap                *unit.Pound
+	BillableWeightJustification      *string
+	NTSRecordedWeight                *unit.Pound
+	ShipmentType                     MTOShipmentType
+	Status                           MTOShipmentStatus
+	Diversion                        bool
+	RejectionReason                  *string
+	Distance                         *unit.Miles
+	Reweigh                          *Reweigh
+	UsesExternalVendor               bool
+	StorageFacility                  *StorageFacility
+	StorageFacilityID                *uuid.UUID
+	ServiceOrderNumber               *string
+	TACType                          *LOAType
+	SACType                          *LOAType
+	PPMShipment                      *PPMShipment
+	CreatedAt                        time.Time
+	UpdatedAt                        time.Time
+	DeletedAt                        *time.Time
+}
+```
+
+Handlers would be in charge of converting payloads into the MTOShipmentUpdate struct and passing it to the UpdateShipment service. The UpdateShipment service would be responsible for fetching the database model (MTOShipment) and updating it appropriately based on the `Present` fields in the MTOShipmentUpdate.
```

Comment:
I guess this is roughly what payload to model functions are doing currently but I don't think we want additional logic added into our handlers.  I think this might add another layer of confusion because not all fields are updatable by each API (customer, office, prime, ...).  Are you saying each handler will provide an API specific **intermediate** structs that calls a dedicated service method that restricts unintended fields being set/updated?

Reply:
In this proposal, the MTOShipmentUpdate struct pretty much takes the place of the MTOShipment model in these PayloadToModel functions that the handlers use, and they can all share this intermediate struct without increasing the complexity of each handler by much at all.

Reply:
>they can all share this intermediate struct without increasing the complexity of each handler by much at all.

can you clarify this section in the ADR a bit more to explain that this wouldn't be adding additional logic added to our handlers?